pkg/generators/forwarding/fluentd/syslog_conf.go

```diff
@@ -1,14 +1,199 @@
 package fluentd
 
+import (
+	"fmt"
+	"regexp"
+	"strings"
+)
+
+// The Syslog output fields can be set to an expression of the form $.abc.xyz
+// If an expression is used, its value will be taken from corresponding key in the record
+var keyre = regexp.MustCompile(`^\$(\.[[:word:]]*)+$`)
```

Comment:
I like $ as the key character - safe and widely used as a "variable expansion" indicator. 
I was thrown at first by the '.' after the $, but I guess the leading '.' is common convention for field paths in JSON paths etc. Perhaps it could be optional, so $abc.xyz means the same as $.abc.xyz? E.g if it's a top-level field `foo` then  $foo reads a bit more naturally than $.foo - it's subjective, I leave it to your discretion.

Reply:
yes, this is jsonpath syntax. fluentd and plugins already support that.

Avoiding initial `.` seems appropriate we can discuss what syntax we expose, and then convert it internally to a jsonpah syntax. leaving it as jsonpath syntax could also be beneficial.

Reply:
Yes, the more I think of it, just ignore my comment - use strict jsonpath and document as such. It won't surprise anyone who has used jsonpath filters with oc or kubectl, and for those who are surprised there's a well known standard syntax they can refer to.